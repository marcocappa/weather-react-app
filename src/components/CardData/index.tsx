import React from 'react';
import { ICityWeatherData } from '../../common/types';

const CardData = ({ data }: { data: ICityWeatherData }): JSX.Element => (
    <div className="card-data">
        <div>
            <p>
                Temperature: <span>{data?.temp} Celcius</span>
            </p>
            <p>
                Apparent temperature: <span>{data?.app_temp} Celcius</span>
            </p>
            <p>
                Relative humidity: <span>{data?.rh}%</span>
            </p>

            <p>
                Weather description:
                <span>{data?.weather?.description}</span>
            </p>
        </div>
        <div>
            <p>
                Last observation time: <span>{data?.ob_time}</span>
            </p>
            <p>
                Timezone: <span>{data?.timezone}</span>
            </p>
            <p>
                Pressure: <span>{data?.pres} mb</span>
            </p>
            <p>
                Wind Speed: <span>{data?.wind_spd} m/s</span>
            </p>
            <p>
                Wind Direction: <span>{data?.wind_dir} Degrees</span>
            </p>
        </div>
    </div>
);

export default CardData;
