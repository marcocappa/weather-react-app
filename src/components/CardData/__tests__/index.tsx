import React from 'react';
import { shallow } from 'enzyme';
import CardData from '../index';

describe('<CardData />', () => {
    const props = {
        data: {
            temp: 8.9,
            app_temp: 9,
            rh: 89,
            weather: { description: 'Overcast clouds' },
            ob_time: ' 2021-03-10 18:06',
            timezone: 'Europe/London',
            pres: 998.9,
            wind_spd: 2.68,
            wind_dir: 200,
        },
    };
    it('should match snapshot', () => {
        const wrapper = shallow(<CardData {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
