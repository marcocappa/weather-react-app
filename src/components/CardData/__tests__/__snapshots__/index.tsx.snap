// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardData /> should match snapshot 1`] = `
<div
  className="card-data"
>
  <div>
    <p>
      Temperature: 
      <span>
        8.9
         Celcius
      </span>
    </p>
    <p>
      Apparent temperature: 
      <span>
        9
         Celcius
      </span>
    </p>
    <p>
      Relative humidity: 
      <span>
        89
        %
      </span>
    </p>
    <p>
      Weather description:
      <span>
        Overcast clouds
      </span>
    </p>
  </div>
  <div>
    <p>
      Last observation time: 
      <span>
         2021-03-10 18:06
      </span>
    </p>
    <p>
      Timezone: 
      <span>
        Europe/London
      </span>
    </p>
    <p>
      Pressure: 
      <span>
        998.9
         mb
      </span>
    </p>
    <p>
      Wind Speed: 
      <span>
        2.68
         m/s
      </span>
    </p>
    <p>
      Wind Direction: 
      <span>
        200
         Degrees
      </span>
    </p>
  </div>
</div>
`;
